spine-ts/spine-pixi/src/Spine.ts

```diff
@@ -0,0 +1,407 @@
+	end: [trackEntry: TrackEntry];
+	event: [trackEntry: TrackEntry, event: Event];
+	interrupt: [trackEntry: TrackEntry];
+	start: [trackEntry: TrackEntry];
+}
+
+export class Spine extends Container {
+	public skeleton: Skeleton;
+	public state: AnimationState;
+
+	private _debug?: ISpineDebugRenderer | undefined = undefined;
+	public get debug(): ISpineDebugRenderer | undefined {
+		return this._debug;
+	}
+	public set debug(value: ISpineDebugRenderer | undefined) {
+		if (this._debug) {
+			this._debug.unregisterSpine(this);
+		}
+		if (value) {
+			value.registerSpine(this);
+		}
+		this._debug = value;
+	}
+
+	// Each slot is a pixi mesh, by default we just visible=false the ones we don't need. This forces a removeChild and addChild every time we need to show a slot.
+	public removeUnusedSlots: boolean;
+	protected slotMeshFactory: () => ISlotMesh;
+
+	private autoUpdateWarned: boolean = false;
+	private _autoUpdate: boolean = true;
+	public get autoUpdate(): boolean {
+		return this._autoUpdate;
+	}
+	public set autoUpdate(value: boolean) {
+		if (value) {
+			Ticker.shared.add(this.internalUpdate, this);
+			this.autoUpdateWarned = false;
+		} else {
+			Ticker.shared.remove(this.internalUpdate, this);
+		}
+		this._autoUpdate = value;
+	}
+
+	private meshesCache = new Map<Slot, ISlotMesh>();
+
+	private static vectorAux: Vector2 = new Vector2();
+	private static clipper: SkeletonClipping = new SkeletonClipping();
+
+	private static QUAD_TRIANGLES = [0, 1, 2, 2, 3, 0];
+	private static VERTEX_SIZE = 2 + 2 + 4;
+	private static DARK_VERTEX_SIZE = 2 + 2 + 4 + 4;
+
+	private lightColor = new Color();
+	private darkColor = new Color();
+
+
+	constructor(skeletonData: SkeletonData, options?: ISpineOptions) {
+		super();
+
+		this.skeleton = new Skeleton(skeletonData);
+		const animData = new AnimationStateData(skeletonData);
+		this.state = new AnimationState(animData);
+		this.removeUnusedSlots = options?.removeUnusedSlots ?? false;
+		this.autoUpdate = options?.autoUpdate ?? true;
+		this.slotMeshFactory = options?.slotMeshFactory ?? ((): ISlotMesh => new SlotMesh());
+
+
+		/**
+		  * This is locked behind https://github.com/pixijs/pixijs/issues/8957
+		  * I don't want to make a custom event emitter and do `this.spineEvents.on` because that's just as "far" as `this.state.addListener`
+		  * So, until pixi fixes the custom event system, I'll stick to spine native events. - @miltoncandelero
+
+			this.spineListeners = {
+				complete: (trackEntry) => this.emit("complete", trackEntry),
+				dispose: (trackEntry) => this.emit("dispose", trackEntry),
+				end: (trackEntry) => this.emit("end", trackEntry),
+				event: (trackEntry, event) => this.emit("event", trackEntry, event),
+				interrupt: (trackEntry) => this.emit("interrupt", trackEntry),
+				start: (trackEntry) => this.emit("start", trackEntry),
+			};
+			this.state.addListener(this.spineListeners);
+		*/
+	}
+
+	public update(deltaSeconds: number): void {
```

Comment:
`update` is meant to be called by the user and provide the deltaTime in seconds.

`internalUpdate` is what actually updates the spine state and the autoUpdate connects directly to this. The user shouldn't call this.

The whole idea behind this is that the tick signature for pixi is something like `(deltaFrames, deltaTime)=>void` with `deltaFrames` being "how many frames passed since last frame if we were running at 60fps?". That means that at 60fps, `deltaFrame` is 1, at 30 fps `deltaFrame` is 2, at 144 fps `deltaFrame` is 2.4, etc.  
On top of that, `deltaTime` is in miliseconds.

Due to this, making a "one size fits all" update function is kinda hard. I focused on user ergonomics and going for an `update` function that the user can call with seconds (matching spine state) and internally doing the ugly math to work with the automatic update